state-chain/pallets/cf-ingress-egress/src/tests/boost.rs

```diff
@@ -787,3 +802,152 @@ fn test_add_boost_funds() {
 		}));
 	});
 }
+
+#[track_caller]
+fn turn_safe_mode_on() {
+	<MockRuntimeSafeMode as SetSafeMode<MockRuntimeSafeMode>>::set_code_red();
```

Comment:
Would be good to also check the state before it is updated (in the test below too). Otherwise we might get a false positive if the asserted value is the default one.